apps/cowswap-frontend/src/modules/tradeQuote/utils/quoteUsingSameParameters.ts

```diff
@@ -45,7 +47,7 @@ function removeQuoteMetadata(appData: string | undefined): string | undefined {
 
   if (!decoded) return
 
-  const { metadata: fullMetadata, ...rest } = decoded
-  const { quote: _, ...metadata } = fullMetadata
+  const { metadata: fullMetadata, ...rest } = decoded as LatestAppDataDocVersion
+  const { quote: _, hooks: __, ...metadata } = fullMetadata
```

Comment:
This is important.
Hooks update in appData takes some time and it affects quote fetching checks.
In order to not trigger quote fetching to often, we omit `hooks` data.
We need this because if you change sell token from one with permit support to another one with permit support, it will trigger quotes update when we don't need it.
See `AppDataInfoUpdater`

Reply:
But a hook affects the gas costs we receive from the API, right? more gas, more cost fee in sell tokens

Reply:
Yes, that's true.
We even discussed it in context of Hooks store: https://cowservices.slack.com/archives/C07FQTUKLJC/p1738850238115259?thread_ts=1738354365.050409&cid=C07FQTUKLJC

Anyway, I will revert this for now, since we already have the same behaviour in prod.